znai-server/src/main/java/com/twosigma/znai/server/DocumentationServer.java

```diff
@@ -99,10 +105,20 @@ public HttpServer create() {
                     router.route("/" + params.get("docId") + "/static"), ctx));
         });
 
+        router.delete("/:docId").handler(ctx -> {
+            MultiMap params = ctx.request().params();
+            String docId = params.get("docId");
+            String queryString = ctx.request().query();
+            String actor = getActorFromQueryString(queryString);
+            new FileCleanHandler(docId, actor).handle(ctx.request());
+        });
+
         router.route("/upload/:docId").handler(ctx -> {
             MultiMap params = ctx.request().params();
             String docId = params.get("docId");
-            new FileUploadVertxHandler(vertx, docId, deployRoot).handle(ctx.request());
+            String queryString = ctx.request().query();
+            String actor = getActorFromQueryString(queryString);
+            new FileUploadVertxHandler(vertx, docId, deployRoot, actor).handle(ctx.request());
```

Comment:
Passed `actor` here will be set in stone for as long as `server` runs. Was this the intention?

Reply:
nevermind, deleting comment